source_gen/test/builder_test.dart

```diff
@@ -198,6 +199,80 @@ void main() {
         });
   });
 
+  test('SharedPartBuilder outputs <partId>.g.part files', () async {
+    await testBuilder(
+        new SharedPartBuilder([const UnformattedCodeGenerator()], '.foo',
+            header: ''),
+        {'$_pkgName|lib/a.dart': 'library a; part "a.g.dart";'},
+        generateFor: new Set.from(['$_pkgName|lib/a.dart']),
+        outputs: {
+          '$_pkgName|lib/a.foo.g.part':
+              decodedMatches(contains(UnformattedCodeGenerator.formattedCode)),
+        });
+  });
+
+  test('SharedPartBuilder outputs `part of` files', () async {
+    await testBuilder(
+        new SharedPartBuilder([const UnformattedCodeGenerator()], '.foo',
+            header: ''),
+        {'$_pkgName|lib/a.dart': 'library a; part "a.g.dart";'},
+        generateFor: new Set.from(['$_pkgName|lib/a.dart']),
+        outputs: {
+          '$_pkgName|lib/a.foo.g.part': decodedMatches(contains('part of')),
```

Comment:
hmm, we have more to do here. we definitely _don't_ want the shared part builder outputting `part of` statements, and I guess we'll need to add one to the `CombiningBuilder` output.